EarlGrey/Core/EarlGreyImpl.m

```diff
@@ -92,6 +92,10 @@ - (BOOL)rotateDeviceToOrientation:(UIDeviceOrientation)deviceOrientation
   return [GREYSyntheticEvents rotateDeviceToOrientation:deviceOrientation errorOrNil:errorOrNil];
 }
 
+- (BOOL)shakeDeviceWithErrorOrNil:(__strong NSError **)errorOrNil {
+  return [GREYSyntheticEvents shakeDeviceWithErrorOrNil:errorOrNil];
```

Comment:
do you need to synchronize here?

Reply:
I am not sure. Gut feeling says no, but that’s an emotional response to the tests passing. 

What usecase do you have in mind? Would it behave differently between shake and rotation?

Reply:
Rotation was one of them but I realized we synchronize in rotation API so it returns only after full rotation has taken affect (or failed). The other scenario is that the interface is changing/animating due to a delayed call ( dispatch_after) and shaking the device during the transition may be ignored?

Reply:
This is already handled by the shakeDevice API in SyntheticEvents.m:
https://github.com/LeoNatan/EarlGrey/blob/92e0bbfe70bc0ad0fe7bd2c58fdc6b566eaf2005/EarlGrey/Event/GREYSyntheticEvents.m#L119

Reply:
Ah I missed it...sorry.